frameworks/PHP/symfony/src/Controller/BenchController.php

```diff
@@ -2,108 +2,134 @@
 
 namespace App\Controller;
 
-use Symfony\Bundle\FrameworkBundle\Controller\Controller;
-use Symfony\Bundle\FrameworkBundle\Controller\ControllerTrait;
+use App\Entity\Fortune;
+use App\Entity\World;
+use Exception;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
-use App\Entity\Fortune;
-use App\Entity\World;
+use Symfony\Component\Routing\Annotation\Route;
 
-class BenchController extends Controller
+class BenchController extends AbstractController
```

Comment:
it would probably be faster to not use AbstractController and only inject explicit dependencies as that lowers the number of dependencies. on the other hand, it makes dependencies like doctrine more lazy. so maybe needs measurement.

Reply:
definitely worth a test, i'll try and post results

Reply:
no noticeable difference in performance injecting over `AbstractController`, performance loss even occurs when injecting services in actions

but i've managed to get significant boost ~7% when i split mono-controller into several dedicated simple controllers

Reply:
👍 makes sense